python/ray/ml/train/integrations/lightgbm/lightgbm_trainer.py

```diff
@@ -41,24 +48,78 @@ class LightGBMTrainer(Trainer):
             dataset. All datasets will be transformed by the ``preprocessor`` if
             one is provided. All non-training datasets will be used as separate
             validation sets, each reporting a separate metric.
-        lightgbm_config: LightGBM training parameters. Refer to
-            `LightGBM documentation <https://lightgbm.readthedocs.io/en\
-/latest/Parameters.html>`_ for a list of possible parameters.
+        lightgbm_config: LightGBM training parameters passed to ``lightgbm.train()``.
```

Comment:
Since this is directly passed to lightbm_ray.train, I'm wondering if it would make sense to just have these as `**lightgbm_train_kwargs`. That way, the API specifying configurations for using the LightGBMTrainer is the same as the API for regular lightgbm training.

Reply:
Hmm so the base Trainer interface accepts kwargs, so it won't be deviating from the interface.

For TorchTrainer and TFTrainer, I don't think there's a use case for kwargs? The `train_loop_config` stays as a dict throughout and never gets converted kwargs. And there's no particular schema for `train_loop_config`.

In this case, since we convert it into kwargs anyways, and there is a particular schema (it should match the signature for lightgbm.train), then it might make sense to make these into kwargs.